pcdsdevices/tpr.py

```diff
@@ -0,0 +1,121 @@
+    group = FCpt(EpicsSignal, '{self.prefix}{self.ch}GROUP', kind="omitted")
+    seqcode = FCpt(EpicsSignal, '{self.prefix}{self.ch}SEQCODE', kind="omitted")
+    fixedrate = FCpt(EpicsSignal, '{self.prefix}{self.ch}FIXEDRATE', kind="omitted")
+    count = FCpt(EpicsSignal, '{self.prefix}{self.ch}COUNT', kind="omitted")
+    destmask = FCpt(EpicsSignal, '{self.prefix}{self.ch}DESTMASK', kind="omitted")
+    destmode = FCpt(EpicsSignal, '{self.prefix}{self.ch}DESTMODE', kind="omitted")
+    src = FCpt(EpicsSignal, '{self.prefix}{self.trg}SOURCE', kind="omitted")
+    eventcode = FCpt(EpicsSignal, '{self.prefix}{self.ch}EVCODE', kind="config")
+    eventrate = FCpt(EpicsSignalRO, '{self.prefix}{self.ch}RATE', kind="normal")
+    label = FCpt(EpicsSignal, '{self.prefix}{self.ch}{self.sys}TCTL.DESC', kind="omitted")
+    label2 = FCpt(EpicsSignal, '{self.prefix}{self.ch}{self.sys2}TCTL.DESC', kind="omitted")
+    delay_ticks = FCpt(EpicsSignal, '{self.prefix}{self.trg}TDESTICKS', kind="omitted")
+    delay_taps = FCpt(EpicsSignal, '{self.prefix}{self.trg}TDESTAPS', kind="omitted")
+    delay_setpoint = FCpt(EpicsSignal, '{self.prefix}{self.sys}TDES', kind="omitted")
+    delay_setpoint2 = FCpt(EpicsSignal, '{self.prefix}{self.sys2}TDES', kind="omitted")
+    ns_delay = Cpt(
+        MultiDerivedSignal,
+        attrs=['delay_ticks', 'delay_taps', 'delay_setpoint'],
+        calculate_on_get=_get_delay,
+        calculate_on_put=_put_last,
+    )
+    ns_delay2 = Cpt(
+        MultiDerivedSignal,
+        attrs=['delay_ticks', 'delay_taps', 'delay_setpoint2'],
+        calculate_on_get=_get_delay,
+        calculate_on_put=_put_last,
+    )
+    ns_delay_scan = FCpt(TprMotor, '{self.prefix}', sys='{self.sys}', trg='{self.trg}', kind="omitted")
+    ns_delay_scan2 = FCpt(TprMotor, '{self.prefix}', sys='{self.sys2}', trg='{self.trg}', kind="omitted")
+    polarity = FCpt(EpicsSignal, '{self.prefix}{self.trg}TPOL', kind="config")
+    width_setpoint = FCpt(EpicsSignal, '{self.prefix}{self.sys}TWID', kind="omitted")
+    width_setpoint2 = FCpt(EpicsSignal, '{self.prefix}{self.sys2}TWID', kind="omitted")
+    width_ticks = FCpt(EpicsSignalRO, '{self.prefix}{self.trg}TWIDTICKS', kind="omitted")
+    width = Cpt(
+        MultiDerivedSignal,
+        attrs=['width_ticks', 'width_setpoint'],
+        calculate_on_get=_get_width,
+        calculate_on_put=_put_last,
+    )
+    width2 = Cpt(
+        MultiDerivedSignal,
+        attrs=['width_ticks', 'width_setpoint2'],
+        calculate_on_get=_get_width,
+        calculate_on_put=_put_last,
+    )
+    enable_cmd = FCpt(EpicsSignal, '{self.prefix}{self.ch}{self.sys}TCTL', kind="omitted")
+    enable_cmd2 = FCpt(EpicsSignal, '{self.prefix}{self.ch}{self.sys2}TCTL', kind="omitted")
+
+    tab_whitelist = ['enable', 'disable']
+    tab_component_names = True
+
+    def __init__(self, prefix, *, channel, name, **kwargs):
+        self.sys = 'SYS0_'
```

Comment:
What is the alternative to have sys and sys2 being attrs? I cannot use global variables or class members because FComponents only format the prefix with instance variables

Reply:
You can include them in the PV stem rather than including them in the FCpt braces.